src/email_bridge.rs

```diff
@@ -41,8 +32,14 @@ pub fn request(app: &Config, email_addr: EmailAddress, client_id: &str, nonce: &
 
     let session = session_id(&email_addr, client_id);
 
-    // Generate a 6-character one-time pad.
-    let chars: String = (0..6).map(|_| CODE_CHARS[rand::random::<usize>() % CODE_CHARS.len()]).collect();
+    // Generate a 12-character one-time pad. For display, we split it in two
+    // groups of 6. We can use the unsafe conversion to str, because we stick
+    // to ASCII.
+    let chars_raw: Vec<u8> = (0..8).map(|_| {
+        CODE_CHARS[rand::random::<usize>() % CODE_CHARS.len()]
+    }).collect();
+    let chars = unsafe { from_utf8_unchecked(&chars_raw) };
+    let chars_fmt = [&chars[0..4], &chars[4..8]].join(" ");
```

Comment:
This is generating 8-character codes :)

I benchmarked `from_utf8` and `from_utf8_unchecked` on my four year old ThinkPad, and the safe version is only 12 nanoseconds slower than the unsafe version. I'd rather pay the 12 nanosecond penalty and be able to claim "no unsafe code in Portier." This also protects us if, somehow, something outside of ASCII like a zero-width space somehow sneaks into `CODE_CHARS`.